client/src/components/ItemCard/ItemCardMedia.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from "react";
+import { useTheme } from "@material-ui/core/styles";
+
+function ItemCardMedia({
+  media,
+  height,
+}: {
+  media: string | null | undefined;
+  height?: number;
+}) {
+  const theme = useTheme();
+
+  const [imgStyle, setImgStyle] = useState({});
+
+  // This hook is to wait for image data to load from bytes
+  // then determine which axis to constrain image
+  // if height > width, constrain by width
+  // else, constrain by height
+  useEffect(() => {
+    const imgSrcRef = document.getElementById(
+      "item-card-media-imgSrc"
+    ) as HTMLImageElement;
+    if (imgSrcRef) {
+      const constrainByHeight = imgSrcRef.height <= imgSrcRef.width;
+      setImgStyle({
+        height: constrainByHeight ? height : "auto",
+        width: constrainByHeight ? "auto" : height,
+      });
+    }
+  }, []);
```

Comment:
```suggestion
  }, [imgStyle]);
```

Reply:
Hmm why is the dependency [imgStyle] needed? I've seen the linter complain a few other times too, but my intention and understanding here is that an empty [] dependency results in this hook acting like a componentDidMount call (and here used to initialize imgStyle state)

Reply:
oh, i see if you want to mimick componentDidUpdate then perhaps just `useEffect(()=>{//do something.})`

Reply:
Only need to initialize this once? Hence the no dependency [] array to fire off on first mount rather than every render update